jazz_fingers.gemspec

```diff
@@ -21,7 +21,7 @@ Gem::Specification.new do |gem|
   # Dependencies
   gem.required_ruby_version = '>= 2.0'
   gem.add_runtime_dependency 'awesome_print', '~> 1.6'
-  gem.add_runtime_dependency 'pry', '~> 0.10'
+  gem.add_runtime_dependency 'pry', '>= 0.13.0'
```

Comment:
Hmmm. Does this mean that someone that can't use `pry >= 0.13.0` will have to keep using old version of `jazz_fingers`?

Maybe we can loose this a bit and work on the code to work on both constraints?

Reply:
Makes sense. I'm taking a look at that now. Seems like the current Pry dependency requirement might be incorrect? I'm trying to find the most recent Pry version number that works with the currently release jazz_fingers 5.0.1...

```rb
[8] pry(main)> JazzFingers::VERSION
# => "5.0.1"
[9] pry(main)> Pry::VERSION
# => "0.10.0"

[5] pry(main)> JazzFingers
# => JazzFingers

[6] pry(main)> JazzFingers.configure
# NoMethodError: undefined method `configure' for JazzFingers:Module

[7] pry(main)> JazzFingers.setup!
# NoMethodError: undefined method `setup!' for JazzFingers:Module


```

Reply:
Commit 857320f restores backwards-compatibility to Pry ~> 0.10

85c5e86 refactors to isolate branching and move special casing to a different class — if that's too design-y or if you'd prefer a different class name, let me know.

```rb
# lib/jazz_fingers.rb L25-40 (85c5e865)

def prompt_class
  if Pry::VERSION >= "0.13.0"
    Prompt
  else
    PromptPry012
  end
end

def prompt
  @prompt ||=
    prompt_class.new(
      colored: config.colored_prompt,
      separator: config.prompt_separator,
      application_name: config.application_name
    )
end
```
<sup>[[source](https://github.com/jmromer/jazz_fingers/blob/85c5e865/lib/jazz_fingers.rb#L25-L40)]</sup>

Reply:
Btw an alternative here would be to mix-in functionality by target pry version rather than inherit:

```rb
class Prompt
  if Pry::VERSION >= "0.13.0"
    include PromptPry013
  else
    include PromptPry012
  end
   # . . . 
end
```

Reply:
Here's what that looks like: d2f71b3
Happy to keep either approach or change tack altogether.